### PR TITLE
Pin details cleanup

### PR DESCRIPTION
**Pull request by Doug-Shannon**
Refactor of Pin Details component

### CODE REVIEW

File: src/app/components/pin-details/pin-login-actions/pin-login-actions.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div *ngIf="!isGathering">
+    <button class="btn primary" (click)="redirectToLogin()">Sign In</button>
+    <p>Don't have an account? <a [routerLink]="['/register']">Register Now</a></p>
+</div>
+<div *ngIf="isGathering">
+    <a (click)="redirectToLogin()">Already in this gathering?</a><br />
+    <a (click)="redirectToLogin()">This is me</a>
```
Comment (JoeKer1) [inline src/app/components/pin-details/pin-login-actions/pin-login-actions.html]: I hate this, but it is correct

File: src/app/services/api.service.ts [hunk 2]
```diff
@@ -101,22 +101,28 @@ export class APIService {
...
-        .catch( (err) => Observable.throw(err.json().error) );
+      .map((res: Array<LookupTable>) => {
+        return res;
+      })
+      .catch((err) => Observable.throw(err.json().error));
   }
 
   public getUserData(): Observable<any> {
-    return this.session.get(`${this.baseUrl}api/v1.0.0/finder/pin/contact/${this.session.getContactId()}/false`)
+    debugger;
```
Comment (JoeKer1) [inline src/app/services/api.service.ts]: Ahem

File: src/app/components/pin-details/pin-login-actions/pin-login-actions.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export class PinLoginActionsComponent implements OnInit {
+
+  @Input() isGathering: boolean = false;
+
+  constructor(
+    private loginRedirectService: LoginRedirectService,
+    private router: Router) {}
+
+  public ngOnInit() {
+  }
```
Comment (JoeKer1) [inline src/app/components/pin-details/pin-login-actions/pin-login-actions.component.ts]: Blank?

File: src/app/components/pin-details/pin-header/pin-header.component.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export class PinHeaderComponent implements OnInit {
+
+  @Input() isGathering: boolean = false;
+  @Input() isPinOwner: boolean = false;
+  @Input() firstName: string = "";
+  @Input() lastName: string = "";
+
+  constructor() {}
+
+  public ngOnInit() {
```
Comment (JoeKer1) [inline src/app/components/pin-details/pin-header/pin-header.component.ts]: delete it?

File: src/app/components/pin-details/say-hi/say-hi.component.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  @Input() user: User;
+  @Input() pin: Pin;
+  @Input() isLoggedIn: boolean = false;
+
+  constructor(
+    private pinService: PinService,
+    private loginRedirectService: LoginRedirectService,
+    private router: Router) { }
+
+  public ngOnInit() { }
```
Comment (JoeKer1) [inline src/app/components/pin-details/say-hi/say-hi.component.ts]: blank? also may have to delete implements OnInit

File: src/app/services/api.service.ts (hunk 2 above)
Comment (Doug-Shannon) [inline src/app/services/api.service.ts]: shoot

**Pull request merged**